src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1879,6 +1885,73 @@ private BoundExpression BindCast(CastExpressionSyntax node, DiagnosticBag diagno
             return BindCastCore(node, operand, targetType, wasCompilerGenerated: operand.WasCompilerGenerated, diagnostics: diagnostics);
         }
 
+        private BoundExpression BindIndexExpression(PrefixUnaryExpressionSyntax node, DiagnosticBag diagnostics)
+        {
+            CheckFeatureAvailability(node, MessageID.IDS_FeatureIndexOperator, diagnostics);
+            GetWellKnownTypeMember(Compilation, WellKnownMember.System_Index__ctor, diagnostics, syntax: node);
+
+            return BindUnaryOperator(node, diagnostics);
```

Comment:
>BindUnaryOperator [](start = 19, length = 17)

I think we shouldn't delegate to BindUnaryOperator, but simply handle this operator specially as we do for RangeExpressionSyntax